File: leetcode/p00557/p00557.py
class Solution:
    # Time Complexity: O(n^2)
    # Space Complexity: O(n)
    def reverseWords(self, s: str) -> str:
        return ' '.join([word[::-1] for word in s.split()])

if __name__ == '__main__':
    s = Solution()

    assert s.reverseWords("Let's take LeetCode contest") == "s'teL ekat edoCteeL tsetnoc"
    assert s.reverseWords("God Ding") == "doG gniD"

File: leetcode/p00557/p00557.go
package p00557

import "strings"

// Time Complexity: O(n^2)
// Space Complexity: O(n)
func reverseWords(s string) string {
	var res strings.Builder

	strs := strings.Split(s, " ")
	for i, str := range strs {
		for j := len(str) - 1; j > -1; j-- {
			res.WriteByte(str[j])
		}

		if i != len(strs)-1 {
			res.WriteRune(' ')
		}
	}

	return res.String()
}
